func.go

```diff
@@ -59,6 +61,10 @@ func RegisterLibFunc(fptr interface{}, handle uintptr, name string) {
 // it will be expanded into a call to the C function as if it had the arguments in that slice.
 // This means that using arg ...interface{} is like a cast to the function with the arguments inside arg.
 // This is not the same as C variadic.
+//
+// There are some limitations when using RegisterFunc on Linux. First, there is no support for function arguments.
+// Second, float32 and float64 arguments and return values do not work when CGO_ENABLED=1. Otherwise, Linux
+// has the same feature parity as Darwin.
```

Comment:
Is there a difference between Darwin and Windows here?

Reply:
Windows doesnt implement RegisterFunc since we just use windows.Syscall on that platform. Otherwise, we'd have to copy all the windows files over into purego and make some modifications to match Darwin and Linux. Doable but seemed like double work to me since I don't think RegisterFunc is really needed for windows.

Reply:
OK, we also want a comment that RegisterFunc doesn't work on Windows.